### DISCUSSION TITLE
Super Mario RPG ROM Header corrupt or missing HW5

### DISCUSSION THREAD

**Discussion by yukikiru** [sanni/cartreader#739] [Need help]
So I was able to get my cart reader built, everything seems okay but I can't get Super Mario RPG (and I assume any other SA1 game) to work. I chcked my snesCIC and clock gen cards to make sure they were soldered okay and they look fine and I calibrated the clockgen as close as I could get it but still getting:

ERROR
ROM HEADER CORRUPT OR MISSING

Everything else is fine, just Mario RPG is causing me trouble.

**Comment by nsx0r** [comment]
It's working fine on HW5 + snesCIC, so it's most likely a contact issue (especially since SA1 games have all 62 pins populated), make sure to clean the pins! Even if they look clean, clean them again.

![image](https://user-images.githubusercontent.com/13054953/226818716-7b7d5619-a81e-4b86-bbdc-fe206490aea9.png)

**Reply by yukikiru** [reply]
Thanks for the reply. I tried to do a very thorough cleaning and got another SA-1 game to test but getting the same thing with it as well (tried a couple at the store and couldn't get any of them to read either)

![PXL_20230322_235457246.jpg](https://user-images.githubusercontent.com/25606505/227064610-77482f74-3580-4087-aa37-e2214a88306b.jpg)

**Reply by yukikiru** [reply]
So after some playing around I was able to get the copy of PGA tour 96 I have to work but not the Copy of super Mario RPG. It works fine in my SNES but the reader may just be super picky about my copy of the game.

**Reply by Chomemel** [reply]
It's the same here. Some SA-1 games are pickier than other.

**Reply by VamPikmin** [reply]
I've tried Shining Scorpion, JkkyoJikkyō Oshaberi Parodius and Panic Bomber W. 
They all show the same error as above.

**Reply by cmcm711** [reply]
Whenever I get this error, replacing the clockgen board resolves the issue. Sometimes it's a faulty CIC chip instead.

**Reply by VamPikmin** [reply]
I had the black screen with the clockgen everytime I picked super nintendo from the menu, which I've replaced, I was able to dump an NP cart. Would that not confirm it's working correctly?
I'm not sure if my supercic has got the correct hex file, these have been flashed years ago and I have been using them in repros.

**Reply by cmcm711** [reply]
I'll give an example of a worst-case situation I've run into more than once while building OSCR HW5 (I've built almost 300 to date, for reference):
- Run self-test, reports "clockgen not found"
- Replace clockgen module, self-test reports all OK, attempt to run clockgen calibration inside SNES menu, device hangs
- Replace clockgen module again, clockgen calibration menu works properly, error "ROM header missing or corrupt" shows when attempting to dump SA1 SNES game
- Replace clockgen module again, SA1 game is successfully identified, but returns incorrect CRC check after dumping; no amount of calibration or cleaning results in a successful dump
- Replace clockgen module again, SA1 game dumps successfully after a few tries

This specific situation is a very rare case, but having one or more of the errors listed above is not uncommon. Also, for what it's worth, I've never had trouble dumping an N64 game successfully using a calibrated clockgen, even modules that show the "ROM header missing or corrupt" error on SA1 SNES games. Maybe N64 dumping is less strict on the clock signal?

It's also worth mentioning that the only errors I've experienced from a malfunctioning CIC module (less common than faulty clockgen modules) have been the "ROM header missing or corrupt" error (not common), or an incorrect CRC check after dumping (more common). I should also indicate that every single CIC module I have built had chips I purchased and had preprogrammed from BuyICNow. I've had ~10-15 faulty chips in almost 300 builds, so not terrible IMO.

**Reply by VamPikmin** [reply]
Thanks for the insight, i have a few more clockgen's that i bought from the same seller, i could try them to see what happens

**Reply by VamPikmin** [reply]
Well I removed the cic module completely and was able to dump Shining Scorpion, Marvelous and Super Mario RPG.
Bomber W and Parodius still give me the same error
I thought the cic was required?

**Comment by splash5** [comment]
I recently try dumping "Kirby 3" and "Kirby Super Deluxe" and got the same problem.
But after I change my power source from a monitor USB port to a external power bank, both game can be recognized and dumped.
I can confirm that I can dump "Super Mario RPG" with that monitor USB port, so maybe this is just only because not enough current
to power some SA1 chip?
Maybe we should measure current when dumping SA1 chip to figure out how much current is needed.

**Reply by Ancyker** [reply]
I've checked:

3V idle (no cart): 130 mA
5V idle (no cart): 150 mA
3V idle (SNES cart): 150 mA
5V idle (SNES cart): 190 mA (230 mA initially, probably caps charging)
dumping SNES: 210 mA
3V idle (SNES SA1 cart): 150 mA
5V idle (SNES SA1 cart): 230 mA (250 mA initially, probably caps charging)
dumping SNES SA1: 250 mA (occasionally hitting 260mA. My meter only has a 10 mA resolution, so it's probably 254-256 and flipping between rounding up and down)

So yes, uses quite a bit more power. Per USB 2.0 spec, ports are only required to supply between 100 mA and 500 mA. Based on that, it is possible to encounter a port that won't power the cart reader at all.

**Reply by Ancyker** [reply]
Decided to try dumping some other systems for comparison:

N64: 140 mA
Game Gear: 150 mA
NES: 190 mA

Note: I forgot I had the 3V3FIX enabled from testing before, so 3.3V ratings are also @ 8MHz. Realized when I went to dump the N64 game and it took forever 😅
